### PR TITLE
add singleton image loader, remove unused libs, increase disk & decrease memory cache

### PR DESCRIPTION
**Pull request by IndusAryan**
annotation processors are now needed if we shift to room sqlite or DI, kapt & ksp were for glide so as transformation lib

### CODE REVIEW

File: app/src/main/java/com/lagradost/cloudstream3/utils/ImageModuleCoil.kt [hunk 1]
```diff
@@ -40,23 +41,23 @@ object ImageLoader {
...
         return ImageLoader.Builder(context)
-            .crossfade(true)
+            .crossfade(250)
             .respectCacheHeaders(false)
             /** !Only use default placeholders and errors, if not using this instance for local
              * image buttons because when animating this will appear or in more cases **/
             //.placeholder(R.drawable.logo)
             //.error(R.drawable.logo)
-            .allowHardware(true)
+            .allowHardware(false) // takes a toll on battery (only allow if app is like instagram or photos)
```
Comment (fire-light42) [inline app/src/main/java/com/lagradost/cloudstream3/utils/ImageModuleCoil.kt:50]: Is this really the case? I thought hardware bitmaps enabled the hardware to save energy by "offloading the work to dedicated co-processors". Can you link where you got this information or explain more in detail.

Comment (IndusAryan) [reply app/src/main/java/com/lagradost/cloudstream3/utils/ImageModuleCoil.kt:50]: the theory is hw decoding works well only in devices that are recent maybe like sdk int > 28, and it is only needed to render large bitmaps.
actually we are fucked instacart released coil 3 in which we dont need our own singleton instead we should inherit their factory in application class

**Review by fire-light42 — CHANGES_REQUESTED** [review]
This does not work on image resources, like the source profiles.

**Comment by IndusAryan** [comment]
fixed load bitmap from drawable, and for the hardware thing, we got software render error, because coil themselves have to solve that bitmap render issue on unsupported device

**Comment by IndusAryan** [comment]
@fire-light42 should we use okhttp client from requests helper here instead of building?

**Comment by fire-light42** [comment]
> @fire-light42 should we use okhttp client from requests helper here instead of building?

Yes

**Comment by IndusAryan** [comment]
i tried to add it, the base client is very confusing, do we set there for each network request, i hope this works same

**Review by fire-light42 — APPROVED** [review]
I do not see any obvious issues, so I will merge this after testing 👍

**Pull request merged**